mlrun/serving/states.py

```diff
@@ -1191,6 +1193,25 @@ def load(self) -> None:
                 raise_missing_schema_exception=False,
             )
 
+        # check if the relevant predict method is implemented when tring to initilaze the model
+        want_async = (
+            self._execution_mechanism == storey.ParallelExecutionMechanisms.asyncio
+        )
+        # Check for async model
+        if want_async:
```

Comment:
why do you need the additional `want_async` variable (it has only a single use)? and if you do I suggest to name it `is_async_enabled`
```suggestion
        if self._execution_mechanism == storey.ParallelExecutionMechanisms.asyncio:
```